client/blocks/upload-image/index.jsx

```diff
@@ -231,10 +234,12 @@ class UploadImage extends Component {
 
 		const isEditingDefaultImage = defaultImage && selectedImage === defaultImage.URL;
 
-		const media = isEditingDefaultImage ? defaultImage : {
-			src: selectedImage,
-			file: selectedImageName
-		};
+		const media = isEditingDefaultImage
+			? defaultImage
+			: {
```

Comment:
I just went through several create/edit/remove session and no side-effects from not resetting `selectedImage`, `selectedImageName`, and `editedImage` were observed. So we can leave it as-is for now. It may affect later changes but logic is simple enough to figure things out.